src/org/skypro/skyshop/search/SearchEngine.java

```diff
@@ -0,0 +1,44 @@
+package org.skypro.skyshop.search;
+
+public class SearchEngine {
+    private final Searchable[] items;
+    private int count;
+
+    public SearchEngine(int size) {
+        this.items = new Searchable[size];
+        this.count = 0;
+    }
+
+    public void add(Searchable item) {
+        if (count >= items.length) {
+            System.out.println("Массив поиска заполнен, нельзя добавить больше элементов!");
+            return;
+        }
+        items[count] = item;
+        count++;
```

Comment:
можно чуть сократить
```suggestion
        items[count++] = item;
```